### PR TITLE
Correctly escape paths

### LINKED ISSUE #11896: deleting a large file with a space in it the cmd is not correct quoted
### Environment

- GitExtensions version: 5.0.0.17897
- GIT version: git version 2.42.0.windows.2
- OS version: 23H2 22631.4037
- .NET version: 8.0.7

### Issue description

When deleting a large file with a space in it the cmd is not correct quoted:

`C:\GitHub\Workplace>"C:\Program Files\Git\bin\git.exe" filter-branch -f --index-filter "git rm -r -f --cached --ignore-unmatch intune/packages/XL Upload/xl-upload.intunewin" --prune-empty -- -
-all`

### Steps to reproduce

delete a file with a space in the directory 

working syntax could be `"C:\Program Files\Git\bin\git.exe" filter-branch -f --index-filter "git rm -r -f --cached --ignore-unmatch 'intune/packages/XL Upload/xl-upload.intunewin'" --prune-empty --
 --all`

### Did this work in previous version of GitExtensions?

_No response_

### Diagnostics

_No response_

### LINKED ISSUE #11897: 'C:\Program' is not recognized - when running "Find large files"
### Environment

- GitExtensions version: 5.0.0.17897
- GIT version: git version 2.42.0.windows.2
- OS version: 23H2 22631.4037
- .NET version: 8.0.7

### Issue description

When running the large file deletion plugin - the following error occurs for git in "Program Files"
```
>for /F %a IN ('"C:\Program Files\Git\bin\git.exe" for-each-ref --format "%^(refname^)" refs/original/') DO "C:\Program Files\Git\bin\git.exe" update-ref -d %a
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
```

### Steps to reproduce

Run Plugins - Find large files

### Did this work in previous version of GitExtensions?

_No response_

### Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by RussKie**
Fixes #11896
Fixes #11897

image

## Test methodology 

- manual
- unit tests

## Test environment(s) 

- GIT 
- Windows 

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by RussKie** [comment]
The tests are failing because on the CI git is defined as "git" and not as "C:\Program Files\Git\bin\git.exe".

File: src/plugins/FindLargeFiles/FindLargeFilesForm.cs [hunk 1]
```diff
@@ -202,27 +202,39 @@ private IEnumerable<GitObject> GetLargeFiles(float threshold)
             }
         }
 
+        private static string GenerateCommand(SortableObjectsList gitObjects)
+        {
+            StringBuilder sb = new();
+            sb.AppendLine(string.Format("SET gitexe=\"{0}\"", AppSettings.GitCommand));
```
Comment (mstv) [inline src/plugins/FindLargeFiles/FindLargeFilesForm.cs:208]:
```suggestion
            sb.AppendLine($"SET gitexe=\"{AppSettings.GitCommand}\"");
```

File: tests/plugins/UnitTests/FindLargeFiles.Tests/FindLargeFilesFormTests.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace DeleteUnusedBranchesTests
+{
+    [TestFixture]
+    public class FindLargeFilesFormTests
+    {
+        [Test]
+        public async Task GenerateCommand_without_deletions_should_return_expected()
+        {
+            await Verifier.Verify(FindLargeFilesForm.GetTestAccessor().GenerateCommand([]));
```
Comment (mstv) [inline tests/plugins/UnitTests/FindLargeFiles.Tests/FindLargeFilesFormTests.cs:11]: ❔ workaround
```suggestion
            string original = AppSettings.GitCommand;
            try
            {
                AppSettings.GitCommand = @"C:\Program Files\Git\bin\git.exe";
                await Verifier.Verify(FindLargeFilesForm.GetTestAccessor().GenerateCommand([]));
            }
            finally
            {
                AppSettings.GitCommand = original;
            }
```

Comment (RussKie) [reply tests/plugins/UnitTests/FindLargeFiles.Tests/FindLargeFilesFormTests.cs:11]: Yes, it was late last night so I didn't get to that. But this, essentially, is the way to fix the tests.
Thank you

**Review by gerhardol — APPROVED** [review]

**Review by mstv — APPROVED** [review]
Looks sensible, have not run

**Pull request merged**